internal/plugin/data_plane_server.go

```diff
@@ -99,26 +112,30 @@ func (*DataPlaneServer) Subscriptions() []string {
 	}
 }
 
-func (dps *DataPlaneServer) run(_ context.Context) {
+func (dps *DataPlaneServer) run() {
```

Comment:
The context that was being passed here (the `messagePipe`'s context) was already listening to the `SIGINT` and `SIGTERM` signals, as that is the "root" context that is initialized in `app.go`.